Makefile

```
@@ -52,10 +52,28 @@ install:
 doc:
 	dune build $(DUNE_OPTS) @doc
 
+doc-build: doc
+	cp -r docs/base docs/source
+	python3 docs/engine/engine.py
+	cd docs && make html
+	rm -rf docs/source
+	cp -r _build/default/_doc/_html/elpi docs/build/html/
+	cp -r _build/default/_doc/_html/elpi-option-legacy-parser docs/build/html/
+	cp -r _build/default/_doc/_html/highlight.pack.js docs/build/html/
+	cp -r _build/default/_doc/_html/odoc.css docs/build/html/
+
+doc-publish: doc-build
+	rm -rf /tmp/gh-pages
+	cp -r docs/build/html/ /tmp/gh-pages
+	git checkout gh-pages
+	rm -rf * && cp -r /tmp/gh-pages/* ./ && rm -rf /tmp/gh-pages && git add . && git commit -m "Updated gh-pages" && git checkout feature/doc
+#	rm -rf * && cp -r /tmp/gh-pages/* ./ && rm -rf /tmp/gh-pages && git add . && git commit -m "Updated gh-pages" && git push origin gh-pages && git checkout feature/doc
```

```suggestion
	OLD_BRANCH=`git branch --show-current`; \
	git checkout gh-pages && rm -rf * && cp -r /tmp/gh-pages/* ./ && rm -rf /tmp/gh-pages && git add . && git commit -m "Updated gh-pages" && git checkout $$OLD_BRANCH
	echo "uploading: enter to continue, ^C to abort"; read DUMMY; git push origin gh-pages"
```

I think we can improve this further:
- root all the doc in `gh-pages/$GIT_REF`
- fail if `git describe --exact-match` fails (that is, if `GIT_REF` is not a tag)
- unless OLD_BRANCH=master, and in that case `GIT_REF=master`

I don't know how we do it, but in Coq we have the doc of all released versions + master (there is a toggle in the bottom left corner).